### PR TITLE
NatSpec: Implement default inheritance.

### PR DESCRIPTION
**Pull request by Marenz**
refs #8911

### CODE REVIEW

File: libsolidity/interface/CompilerStack.cpp [hunk 1]
```diff
@@ -367,6 +362,11 @@ bool CompilerStack::analyze()
 						if (!contractLevelChecker.check(*contract))
 							noErrors = false;
 
+		DocStringAnalyser docStringAnalyser(m_errorReporter);
+		for (Source const* source: m_sourceOrder)
+			if (source->ast && !docStringAnalyser.analyseDocStrings(*source->ast))
+				noErrors = false;
+
```
Comment (hrkrshnn) [inline libsolidity/interface/CompilerStack.cpp:369]: Why is this moved?

Comment (hrkrshnn) [reply libsolidity/interface/CompilerStack.cpp:369]: I get it now.

File: Changelog.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ Language Features:
 
 Compiler Features:
  * NatSpec: Add fields "kind" and "version" to the JSON output.
+ * NatSpec: Implement default inheritance.
```
Comment (chriseth) [inline Changelog.md]: Inherit tags from unique base if derived function does not provide any.

File: docs/natspec-format.rst [hunk 3]
```diff
@@ -42,10 +42,6 @@ The following example shows a contract and a function using all available tags.
 
 .. note::
 
-  NatSpec currently does NOT apply to public state variables (see
```
Comment (chriseth) [inline docs/natspec-format.rst:45]: Has this been fixed already?

File: docs/natspec-format.rst [hunk 4]
```diff
@@ -60,7 +56,6 @@ The following example shows a contract and a function using all available tags.
     /// @notice You can use this contract for only the most basic simulation
     /// @dev All function calls are currently implemented without side effects
     contract Tree {
-        /// @author Mary A. Botanist
```
Comment (chriseth) [inline docs/natspec-format.rst:63]: Did this not compile with autor?

File: libsolidity/analysis/DocStringAnalyser.cpp [hunk 5]
```diff
@@ -79,6 +103,13 @@ bool DocStringAnalyser::visit(VariableDeclaration const& _variable)
 				"Documentation tag @title and @author is only allowed on contract definitions. "
 				"It will be disallowed in 0.7.0."
 			);
+
+	if (
+		_variable.annotation().docTags.empty() &&
+		_variable.annotation().baseFunctions.size() == 1
+	)
+		copyMissingTags(_variable.annotation(), (*_variable.annotation().baseFunctions.begin())->annotation());
```
Comment (chriseth) [inline libsolidity/analysis/DocStringAnalyser.cpp]: If you `dynamic_cast` inside `copyMissingTags` anyway, you could also move the `->annotation()` call inside `copyMissingTags`. Maybe even make the `baseFunctions` the parameter and then you can also move the check for `baseFunctions.size() == 1`

**Comment by chriseth** [comment]
Looks good apart from the small style suggestion!

File: docs/natspec-format.rst (hunk 3 above)
Comment (Marenz) [reply docs/natspec-format.rst:45]: I tested it and it was working, I think the person who fixed it just forgot to remove this

File: docs/natspec-format.rst (hunk 4 above)
Comment (Marenz) [reply docs/natspec-format.rst:63]: It did, but this PR also adds deprecation for the `author` field for functions, I figured we don't want a deprecated field in the example.

File: libsolidity/analysis/DocStringAnalyser.cpp (hunk 5 above)
Comment (Marenz) [reply libsolidity/analysis/DocStringAnalyser.cpp]: I can do that only partially. I still need to do the `.size() == 1` check for the callable function as we also check the name&parameter types first.

**Review by chriseth — APPROVED** [review]
